auth0/data_source_auth0_client.go

```diff
@@ -10,17 +10,20 @@ import (
 )
 
 func newDataClient() *schema.Resource {
-	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
-	delete(clientSchema, "client_secret_rotation_trigger")
-
-	addOptionalFieldsToSchema(clientSchema, "name", "client_id")
-
 	return &schema.Resource{
 		Read:   readDataClient,
-		Schema: clientSchema,
+		Schema: newClientSchema(),
 	}
 }
 
+func newClientSchema() map[string]*schema.Schema {
+	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
+	delete(clientSchema, "client_secret_rotation_trigger")
+	delete(clientSchema, "client_secret")
```

Comment:
@sergiughf With our decision to remove `client_secret` from the global client schema, it also makes sense to remove for all clients. I don't believe we've released that DS yet so this should be fine. Can always add in if someone has a compelling use case for it.